### ISSUE TITLE
Cache kept in indexing step

### ISSUE THREAD

**Issue by StyXman** [openstreetmap/osm2pgsql#437]
I noticed that `osm2pgsql` version 0.88.0 keeps the cache when it has finished processing the ways and relations and starts indexing the tables. Is it needed? Could that memory be freed to the kernel's page cache?

**Comment by pnorman** [comment]
Yes, freeing memory once out of pending would be good, since if the DB is on the same machine as osm2pgsql it would give more memory for index creation.

**Comment by hholzgra** [comment]
I actually spent a bit of time on this earlier today

https://github.com/hholzgra/osm2pgsql/commit/36c397f3e694075d0a74371fb3cc0a21819a3aaf

Maybe I should have checked pending open issues and pull requests earlier ...

**Comment by hholzgra** [comment]
So far I've just added freeing of mid/out clones as part of the process_polygons method in middle-pgsql.cpp, the mid clones hold copies of the cache auto_ptr, so these have to be cleaned up to bring the reference counter down to 1, and eventually 0, in the final middle_pgsql::stop()
